### ISSUE TITLE
IFrame remains visible if one of parent container is invisible

### ISSUE THREAD

**Issue by nicoulaj** [flex-users/flex-iframe#71]
Originally filed by vmk.gr...@gmail.com on 2009-05-04T12:48:31

What steps will reproduce the problem?
1. Put IFrame inside any Container
2. Toggle visibility of the Container programmatically

What is the expected output? 
IFrame is visible when all parent containers are visible,
IFrame is invisible when any parent container is invisible,

What do you see instead?
IFrame remains visible all the time

What version of the product are you using? On what operating system?
1.3.3

Please provide any additional information below.
To fix this define event listeners for SHOW and HIDE events for each parent
Container.

**Comment by nicoulaj** [comment]
Updated by alistair.rutherford on 2009-05-13T20:21:46

Please read the FAQ carefully. It sounds like you have not setup the WMode=Opaque 
issue properly.

If you examine the demos you will see the frame inside a tab navigator container and 
the iframe contents are switched out of visibility as the tabs are switched. If the WMode=Opaque setting hasn't been setup the frame contents stay visible. I just want 
to double check before I accept this as an defect.

**Comment by nicoulaj** [comment]
Updated by vmk.gr...@gmail.com on 2009-05-19T10:52:59

Strange, but I'm using default WMode (Maybe it is already Opaque? According to the
article it should be Window). Everything works fine including TabNavigators and
ViewStacks except the case when one of IFrame's parent containers made invisible
programmatically, i.e.
container.visible = false
So, as I said we should listen not only CHANGE event but also SHOW and HIDE.

**Comment by nicoulaj** [comment]
Updated by alistair.rutherford on 2009-05-21T09:39:02

My apologies you are quite correct. I obviously didn't read your description
properly. I will need to go away and try this out.

**Comment by nicoulaj** [comment]
Updated by Julien.N...@gmail.com on 2009-10-07T11:46:07

Original ticket set status to Accepted (we converted to open)

**Comment by nicoulaj** [comment]
Updated by andr....@gmail.com on 2009-12-01T14:12:20

I've attached a patch to fix the issue

**Comment by nicoulaj** [comment]
Updated by Julien.N...@gmail.com on 2009-12-05T14:17:48

I think there is no need to define a new "parentVisibility" property, that is the
role of "validForDisplay" property.

**Comment by nicoulaj** [comment]
Updated by andr....@gmail.com on 2009-12-06T12:31:59

I've added parentVisibility property to separate these checks. If we'll utilize
validForDisplay property for both checks (show/hide and index change) it will be
required to check index validity and visibility of parent containers on each event:
show/hide and change. I realize that it is not very significant affect, but for
performance it will be better to have two different variables for this... Maybe just
rename validForDisplay to something like indexValid?
